src/dynarec/rv64/dynarec_rv64_00.c

```diff
@@ -723,8 +723,35 @@ uintptr_t dynarec64_00(dynarec_rv64_t* dyn, uintptr_t addr, uintptr_t ip, int ni
             GETED(0);
             emit_test32(dyn, ninst, rex, ed, gd, x3, x4, x5);
             break;
+        case 0x86:
+            INST_NAME("(LOCK)XCHG Eb, Gb");
+            nextop = F8;
+            if(MODREG) {
+                GETGD;
+                GETED(0);
+                MV(x1, gd);
+                ANDI(gd, gd, ~0xff);
```

Comment:
the xh part can also be exchanged. I don't think this is handled in this code.

Reply:
Oh yes.